Create "Integrate UI" related guides - Closes 1708 (#1715)
LiskHQ/lisk-docs: docs/modules/ROOT/pages/integrate-blockchain/integrate-UI/index.adoc

* The UI should have a page to transfer tokens to any account on the sidechain.
* The UI should also provide a faucet screen, where an account can be seeded with tokens.
* The UI should also inform the user about any errors that might occur during the aforementioned processes.
====

TIP: The relevant files & folders discussed in this guide are the {url_api}[api.js], the {url_package}[package.json], and the {url_frontend}[hello_frontend], respectively.

== Preliminaries and directory structure
Our goal in this chapter is to focus on the integration of the Lisk client with a UI.
To jump-start with the integration directly, we have created a {url_boilerPlate}[boilerplate UI] project for you.

Throughout this guide, we will build on top of the aforementioned template and integrate the <<hello, features listed>> earlier into the Hello sApp.

.Prerequisites
[NOTE]
====
To use this guide, it is assumed that the following criteria have been met:

. The `hello_client` has been created along with the functionalities mentioned in the xref:{url_helloClient_overview}["Hello World" client] section. 

> . The UI template: `hello_frontend` which is based on react and semantic-UI has been created as described in the boilerplate's {url_helloapp_readMe}[README.md] file.

Author: ```suggestion
. The UI template: `hello_frontend` which is based on the react and semantic-UI has been created as described in the boilerplate's {url_helloapp_readMe}[README.md] file.
```

. The UI template based on the react and semantic-UI has been created as described in the boilerplate's {url_helloapp_readMe}[README.md] file.

> ====

Author: > Also it would be good to include the steps to set up the folder structure, like downloading the UI boilerplate and client, installing the dependencies, and starting boilerplate/ client to verify successful setup.

The folder structure and how to set up the project for the Hello sApp are already described [here](https://liskhq.github.io/lisk-docs/beta/build-blockchain/create-sidechain-app.html#initializing-a-new-sidechain-application) and in the boiler plate's [README.md](https://github.com/LiskHQ/lisk-sdk-examples/blob/1582-UI-boiler-plate/guides/07-ui-boilerplate/hello_frontend/README.md) and are mentioned as pre-requisites.

My initial consideration was to download the boilerplate `hello_frontend` folder directly, but you can't do it with a simple command and it is kinda complex, so I created the boilerplate guide to ease things for the reader.

Starting boilerplate/client-related steps are already present in the guide.

Will update the pre-requisites as suggested.

Once all the aforementioned pre-requisites have been met, the `hello` directory should look like the following:

.hello/
----
hello/
├── hello_client/
└── hello_frontend/
    ├── public/
    ├── src/
    │    ├── components/ <1>
    │    │    ├── faucet.js
    │    │    ├── getAccountDetails.js
    │    │    ├── getHello.js
    │    │    ├── home.js 
    │    │    ├── messageTimeline.js
    │    │    ├── newAccount.js
    │    │    ├── sendHello.js
    │    │    └── transfer.js
    │    ├── layout/
    │    │    └── header.js
    │    ├── App.css
    │    ├── App.js
    │    ├── App.test.js
    │    ├── index.css
    │    ├── index.js
    │    ├── logo.png
    │    ├── reportWebVitals.js
    │    └── setupTest.ts
    ├── package-lock.json 
    └── package.json
----

<1> Contains all the web pages, that will provide the  <<hello, aforementioned functionalities>>.

== Installing dependencies

> With the UI template ready, we now need to install the `@liskhq/lisk-client` package. Through `lisk-client`, the UI will communicate with the *hello_client*.

Author: ```suggestion
Now, navigate to the `hello_frontend` folder and install the `@liskhq/lisk-client` package.
Through `lisk-client`, the UI will communicate with the *hello_client*.
```
